bndtools.core/src/bndtools/views/repository/RepositoriesView.java

```diff
@@ -443,6 +460,31 @@ public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
         }.schedule();
     }
 
+    private void configureOfflineAction() {
+        Workspace workspace = Central.getWorkspaceIfPresent();
```

Comment:
Ok i see now, i didn't read the full code, and I see that this method is not called during the initial painting of the UI in createPartControl() so it wouldn't block, carry on then.